addons/web_editor/static/tests/html_field_tests.js

```diff
@@ -36,13 +124,199 @@ QUnit.module("WebEditor.HtmlField", ({ beforeEach }) => {
+                            string: "Header",
+                            type: "html",
+                            required: true,
+                        },
+                        body: {
+                            string: "Message",
+                            type: "html"
+                        },
+                    },
+                    records: [],
+                }
             },
         };
         target = getFixture();
 
         setupViewRegistries();
     });
 
+    QUnit.module('basic');
+
+
+    QUnit.test('simple rendering', async function (assert) {
+        assert.expect(2);
+        serverData.models['note.note'].records = [{
+            id: 1,
+            display_name: "first record",
+            header: "<p>  &nbsp;&nbsp;  <br>   </p>",
+            body: "<p>toto toto toto</p><p>tata</p>",
+        }];
+
+        await makeView({
+            type: "form",
+            resId: 1,
+            resModel: "note.note",
+            serverData,
+            arch: `
+                <form>
+                    <field name="body" widget="html" style="height: 100px"/>
+                </form>`,
+        });
+
+        const field = target.querySelector('.o_field_html[name="body"]');
+        const editable = field.querySelector("[contenteditable='true']");
+        console.log(target.cloneNode(true))
+        assert.strictEqual(editable.innerHTML,
+            '<p>toto toto toto</p><p>tata</p>',
+            "should have rendered a div with correct content");
+        assert.strictEqual(field.getAttribute('style'), 'height: 100px',
+            "should have applied the style correctly");
+    });
+
+    QUnit.test('check if required field is set', async function (assert) {
+        assert.expect(3);
+
+        serverData.models['note.note'].records = [{
+            id: 1,
+            display_name: "first record",
+            header: "",
+            body: "<p>toto toto toto</p><p>tata</p>",
+        }];
+
+        await makeView({
+            type: "form",
+            resId: 1,
+            resModel: "note.note",
+            serverData,
+            arch: `
+                <form>
+                    <field name="header" widget="html" style="height: 100px"/>
+                </form>`,
+        });
+        await click(target, '.o_form_button_save');
+        assert.containsOnce(target, ".o_notification");
+        const notif = target.querySelector(".o_notification");
+        assert.strictEqual(
+            notif.querySelector(".o_notification_content").textContent,
+            "Header"
+        );
+        assert.hasClass(notif, "border-danger");
+
+    });
+
+    QUnit.test('colorpicker', async function (assert) {
+        assert.expect(10);
+
+        serverData.models['note.note'].records = [{
+            id: 1,
+            display_name: "first record",
+            header: "<p>  &nbsp;&nbsp;  <br>   </p>",
+            body: "<p>toto toto toto</p><p>tata</p>",
+        }];
```

Comment:
Why aren't all the data at the top of the file like it used to ? This is not good because this test has side effects on other tests since it modified the data that all the tests use. This is why the data is supposed to be put on top of the file.

Reply:
I did the same as all the other tests. I can modify everybody if you want.

Reply:
Yes, I think it is a mistake in the other tests as well.